src/Home/useBluetoothStatus.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from "react"
+import { Platform, AppState } from "react-native"
+import { isBluetoothEnabled } from "../gaen/nativeModule"
+
+export const useBluetoothStatus = (): boolean => {
+  const [btStatus, setBTStatus] = useState(false)
+  useEffect(() => {
+    const fetchBTEnabled = async () => {
+      const status = await fetchBTStatus()
+      setBTStatus(status)
+    }
+
+    AppState.addEventListener(getOSListener(), () => fetchBTEnabled())
+    return AppState.removeEventListener(getOSListener(), () => fetchBTEnabled())
+  }, [])
+
+  return btStatus
+}
+
+const fetchBTStatus = async (): Promise<boolean> => {
+  return isBluetoothEnabled()
+}
+
+const getOSListener = (): "focus" | "change" => {
+  return Platform.select({
+    ios: "change",
+    android: "focus",
```

Comment:
`focus` is android-only, so it can't be used for `ios`. `focus` and `blur` events account for both foregrounding the app after it has been backgrounded (like `change`), and also re-focusing the app after pulling down the notification bar (where Android users can toggle bluetooth on/off).